### ISSUE TITLE
Translations

### ISSUE THREAD

**Issue by TeamSpen210** [BenVlodgi/BEE2.4#416]
I've added the ability to translate the BEE2, so the text in the interface can be changed to match the language set for your computer. Would anyone be interested in helping do the translations? There's approximately 250 lines, although most of those are single words like 'Music', or 'OK'. Basically you'd download a master translation file, add the translated versions of text, and send it back or pull-request.

**Comment by LautaroL20** [comment]
Okay, spanish is my maternal language, so ill gonna do the spanish translation.

**Comment by Neshiro** [comment]
Hello, dear developpers. I'm French and I'm willing to do the French translation of the BEEMOD2. Would you be interested ?

**Comment by LautaroL20** [comment]
Of course, just get in contact with Spencer @TeamSpen210 and he will guide you thru the process with all the corresponding steps.

**Comment by TeamSpen210** [comment]
Here's the master translation file: [BEE2.pot](https://github.com/BEEmod/BEE2.4/blob/dev/i18n/BEE2.pot)
It's just a text file, you can edit it with anything. `msgid` is the English version, and `msgstr` would then be the translated version. Or you can use the editor I linked in the first comment.

**Comment by Neshiro** [comment]
Hello dear developpers. Here is my translation in French of the BEEMOD2.
I hope it will serve you well.
I'm looking forward to helping you again !

(I will keep a copy of the file if any day you would need it again.)

[BEEMOD2_French.zip](https://github.com/BEEmod/BEE2.4/files/3044675/BEEMOD2_French.zip)

**Comment by TeamSpen210** [comment]
Thank you very much! I've added it ready for the next release. If you put the attached `.mo` file into the BEE2 `i18n` folder ("internationalisation"), it should automatically recognise and switch over to use your translations.
[fr.zip](https://github.com/BEEmod/BEE2.4/files/3045844/fr.zip)

**Comment by TANER05** [comment]
I can help with the Turkish translation.

**Comment by LautaroL20** [comment]
Get in contact with @TeamSpen210 and he will guide you thru the process.

**Comment by Neshiro** [comment]
Hello dear developers.
I've made an updated version of the translation I did last year.

If I may ask, why do I not have every string in the file ? For instance, I have only half of the strings present in the Style and Object Properties window.
[BEEMOD2_French updated.zip](https://github.com/BEEmod/BEE2.4/files/4470261/BEEMOD2_French.updated.zip)

**Comment by TPEcool** [comment]
Can i please do **_Russian_**?

**Comment by TPEcool** [comment]
**_MY MASTER TRANSLATION DOESNT WORK_**

**Comment by Databyte8** [comment]
I can make German

**Comment by CMSEv7** [comment]
I may do an Esperanto translation.

**Comment by Thedoczek** [comment]
I added a [POEditor page](https://poeditor.com/join/project?hash=eRRj4UGEKo) where you can import and translate every new languages that you will want to, because why to create separate for every translation?
And I am doing Polish translation, it's 27% done.
![Image.](https://user-images.githubusercontent.com/69758777/108416244-b8d59800-722e-11eb-8610-f9cf57a3ecee.png)

**Comment by Thedoczek** [comment]
@TeamSpen210 @LautaroL20 I want to launch my game with translations privately, to check are they correct, is it possible

**Comment by LautaroL20** [comment]
You could just make a pull request and TeamSpen210 would do a check for any issues.

**Comment by Thedoczek** [comment]
I don't know how to make pull request

**Comment by Thedoczek** [comment]
@LautaroL20 Should I commit to dev or to master?

**Comment by LautaroL20** [comment]
Yes, dev is fine.

**Comment by TeamSpen210** [comment]
@Thedoczek all you need to do is go to the export, and export it as "gettext .mo". Then put it in the `i18n` folder with the right country/lang code name (check logs to see what it's checking for).

**Comment by CMSEv7** [comment]
What is the situation for an Esperanto translation ?
I had the idea to select the language of BEE2.4 in the app, not use the language of the system.

**Comment by RadRussianRus** [comment]
@Thedoczek Thanks for the POEditor page! I've translated part of the Russian, but sadly, I hit a free limit with it. I've exported it and will edit in text editor though.

**Comment by KaziPlays** [comment]
I can make a Portuguese translation if there's no translation yet.

**Comment by LautaroL20** [comment]
Of course, feel free to make it. Here is the link to the translation project: https://poeditor.com/join/project?hash=eRRj4UGEKo
For aditional steps, you can ask TeamSpen210.

**Comment by KaziPlays** [comment]
Uh... ![Screenshot_20210314-075550](https://user-images.githubusercontent.com/72144613/111065895-f37fc880-849a-11eb-87ca-cf3e541bb82a.png)

**Comment by KaziPlays** [comment]
@TeamSpen210 needs to fix that.

**Comment by ebartaus** [comment]
> hello. i noticed that the russian localization of bee2 contains errors and flaws. i would like to fix it, can i do it? (actually, i'm already translating)

i finished the translation. here it is:
[bee2_russian.zip](https://github.com/BEEmod/BEE2.4/files/6524856/bee2_russian.zip)

**Comment by TeamSpen210** [comment]
@ebartaus Unfortunately it doesn't seem you included the translation file, you'll need to export the `.po` file, `.pot` is the "template" file that includes only the original English.

**Comment by ebartaus** [comment]
> @ebartaus Unfortunately it doesn't seem you included the translation file, you'll need to export the `.po` file, `.pot` is the "template" file that includes only the original English.

oh, i'm sorry. here is the `.po` file
[bee2_russian.zip](https://github.com/BEEmod/BEE2.4/files/6551953/bee2_russian.zip)

**Comment by LautaroL20** [comment]
Message from @Laudanum-dev (https://github.com/BEEmod/BEE2.4/issues/1705): 

"I created the fixed russian translation for beemod: removed unneeded puns in translation, fixed some typos, used terminology from VDC wiki (like the translated names of items, or the translation of word "brush"), and fixed some lines that didn't have translation."

**here is the fixed russian translation download file:** [ru-fix.zip](https://github.com/BEEmod/BEE2.4/files/7126416/ru-fix.zip)

**Comment by KaziPlays** [comment]
I'm back, translating BEE2 to portuguese.

**Comment by igrbml** [comment]
i noticed that the russian translation of bee2 uses hyphens instead of dashes, programmer quotation marks instead of guillemets, etc.

here's the typografed version: [ru.po(.txt)](https://github.com/BEEmod/BEE2.4/files/8734546/ru.po.txt)

**Comment by ARTUTTU** [comment]
> Okay, spanish is my maternal language, so ill gonna do the spanish translation.

thx

**Comment by Lirniri** [comment]
Full translation, completed at 95.31%.
The last texts did not make sense to translate.

This is my first experience of translation, there may be errors.

[BEE2_Russian.zip](https://github.com/user-attachments/files/19594777/BEE2_Russian.zip)
